web_poet/page_inputs/http.py

```diff
@@ -181,6 +182,10 @@ def urljoin(self, url: Union[str, _RequestUrl, _ResponseUrl]) -> _RequestUrl:
         If *url* is relative, it is made absolute relative to :attr:`url`."""
         return _RequestUrl(urljoin(str(self.url), str(url)))
 
+    def fingerprint(self) -> str:
+        """Return the fingerprint of this request."""
+        return sha1(str(self.url).encode()).hexdigest()  # TODO
```

Comment:
it should include method, request body and headers

Reply:
I see this is a placeholder :)

Reply:
> I also wonder if this fingerprinting needs to be user-configurable.

It seems that can be desirable to make it configurable.
For example, we can add `request_fingerprinter` argument to `HttpClient.__init__`. 

The issue though is that we must use the same fingerprinter both during test generation and test running, which means test running is no longer independent from test generation. You can't e.g. reuse Scrapy's fingerprinter in test generation, and then run the tests.

Because of this, it looks like it's somewhat complex to make it configurable in a nice way. So, I'd focus on a reasonable default function for now. 

The goal of this fingerprinter is slightly different from e.g. Scrapy's, as it's not used for deduplication of requests, it's used for identification of requests. So, it seems it's fine for these fingerpints to be more strict.

We'll have issue, e.g. with random hashes, or with timestamps used as parts of URL, but it seems we'll be able to add customization options later; it doesn't look like a requirement for this PR.

Reply:
To make it easier to customize later, I'd make fingerprinting a method of HttpClient, or a standalone function, not a method of HttpRequest.